File: app/src/main/java/ru/e6atb/chat/MessageNotificationPolicy.java
package ru.e6atb.chat;

import java.util.Set;

final class MessageNotificationPolicy {
	private MessageNotificationPolicy() {
	}

	static boolean shouldNotify(boolean bootstrapComplete, boolean sentByMe,
			long messageId, long readAt, Set<Long> readMessageIds) {
		return bootstrapComplete
				&& !sentByMe
				&& readAt <= 0
				&& (readMessageIds == null || !readMessageIds.contains(messageId));
	}
}

File: app/src/main/kotlin/ru/e6atb/chat/MessageNotificationPolicy.kt
package ru.e6atb.chat

object MessageNotificationPolicy {
    @JvmStatic fun shouldNotify(bootstrapComplete: Boolean, sentByMe: Boolean, messageId: Long, readAt: Long, readMessageIds: Set<Long>?): Boolean =
        bootstrapComplete && !sentByMe && readAt <= 0 && (readMessageIds == null || !readMessageIds.contains(messageId))
}
